File: Daily Challenge/129. Sum Root to Leaf Numbers.cs
public class Solution
{
    public int SumNumbers(TreeNode root)
    {
        return DFS(root, 0);
    }

    private int DFS(TreeNode node, int currentSum)
    {
        if (node is null) return 0;

        int newSum = currentSum * 10 + node.val;

        if (node.left is null && node.right is null) return newSum;
        return DFS(node.left, newSum) + DFS(node.right, newSum);
    }
}

File: Daily Challenge/129. Sum Root to Leaf Numbers.py
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        return self.DFS(root, 0)

    def DFS(self, node: Optional[TreeNode], currentSum: int) -> int:
        if node is None:
            return 0

        newSum = currentSum * 10 + node.val
        if node.left is None and node.right is None:
            return newSum

        return self.DFS(node.left, newSum) + self.DFS(node.right, newSum)
